### PR TITLE
Broadcast mult update

### PR DESCRIPTION
**Pull request by joakim-hove**
When MULT? has been updated runtime in the SCHEDULE section we need to broadcast the updated `TransMult` container.

Depends on: https://github.com/OPM/opm-common/pull/2822

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
@akva2

**Comment by akva2** [comment]
mostly stylistic and i won't insist, but i personally would rather do it as a private template 
```
template<class T>
void broadcast(Comm, T&);
```
which you instance for TransMult and Schedule in the cpp file.

This would allow a more uniform interface for users.

**Comment by akva2** [comment]
almost.

```
template<class T>
broadcast(Parallel::Communication comm, T& data)
{
    Opm::EclMpiSerializer ser(comm);
    ser.broadcast(data);
}

template broadcast<TransMult>(Comm, TransMult&);
template broadcast<TransMult>(Comm, TransMult&);
```

less dupe and easy instancing for more types.

File: opm/simulators/utils/ParallelSerialization.hpp [hunk 1]
```diff
@@ -28,6 +28,7 @@ class Schedule;
 class SummaryConfig;
 class UDQState;
 class WellTestState;
+class TransMult;
```
Comment (akva2) [inline opm/simulators/utils/ParallelSerialization.hpp:31]: no longer needed.

File: opm/simulators/utils/ParallelSerialization.cpp [hunk 2]
```diff
@@ -57,9 +57,15 @@ void eclStateBroadcast(Parallel::Communication comm, EclipseState& eclState, Sch
...
+void broadcast(Parallel::Communication comm, T& data)
 {
     Opm::EclMpiSerializer ser(comm);
-    ser.broadcast(schedule);
+    ser.broadcast(data);
 }
+
+
+template broadcast<TransMult>(Comm, TransMult&);
+template broadcast<TransMult>(Comm, Schedule&);
```
Comment (akva2) [inline opm/simulators/utils/ParallelSerialization.cpp:70]: Schedule template param.

**Comment by akva2** [comment]
yeah, i'd probably do eclBroadcast ?

**Comment by joakim-hove** [comment]
> i'd probably do eclBroadcast ?

That was on tongue as well.

**Comment by joakim-hove** [comment]
jenkins build this please

**Pull request merged**